blockchain/blockchain.go

```diff
@@ -589,6 +605,19 @@ func blockByHash(txn db.Transaction, hash *felt.Felt) (*core.Block, error) {
 	})
 }
 
+func StoreL1HandlerMsgHashes(dbTxn db.Transaction, blockTxns []core.Transaction) error {
+	for _, txn := range blockTxns {
+		if l1Handler, ok := (txn).(*core.L1HandlerTransaction); ok {
+			l1HandlerTxnHashBytes := txn.Hash().Bytes()
+			err := dbTxn.Set(db.L1HandlerTxnHashByMsgHash.Key(l1Handler.MessageHash()), l1HandlerTxnHashBytes[:])
```

Comment:
Do you know why the `MessageHash()` doesn't return a `*felt.Felt`?

Also, instead of the following:

```go
l1HandlerTxnHashBytes := txn.Hash().Bytes()
err := dbTxn.Set(db.L1HandlerTxnHashByMsgHash.Key(l1Handler.MessageHash()), l1HandlerTxnHashBytes[:])
```

I think it should be 

```go
err := dbTxn.Set(db.L1HandlerTxnHashByMsgHash.Key(l1Handler.MessageHash().Marshal()), tx.Hash().Marshal())
```

I am checking why MessageHash() is not a felt? I think we can make it a felt.

Reply:
I'm not sure why MessageHash() doesn't return a felt, although it looks like there was never a need for it to. I think changing it is outside the scope of this PR though.

I implemented your suggestion though